troposphere/ec2.py

```diff
@@ -757,13 +757,47 @@ class LaunchSpecifications(AWSProperty):
     }
 
 
+class SpotFleetLaunchTemplateSpecification(AWSProperty):
```

Comment:
This is the same as LaunchTemplateSpecification in the same file -- I wasn't sure whether it's better to create a new one, or use the existing one in a new context (in the CF documentation, all these sub-properties have their own identity in each context, even if otherwise identical.)

Reply:
I would suggest using SpotFleetLaunchSpecification and then changing the LaunchSpecifications entry to use it. I'm saying this based on what is in the "specification" files.

Reply:
Hey Mark,

Not sure I understand (part of the difficulty in discussing these things is the inconsistencies in naming all across CloudFormation, particularly with plurals!)

The CloudFormation change this PR is intended to support adds a new [LaunchTemplateConfigs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-spotfleet-spotfleetrequestconfigdata.html#cfn-ec2-spotfleet-spotfleetrequestconfigdata-launchtemplateconfigs) property to SpotFleetRequestConfigData, in addition to the LaunchSpecifications property that was already there. So the change doesn't actually relate to LaunchSpecifications.

Each [LaunchTemplateConfig](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-spotfleet-launchtemplateconfig.html) contains a [LaunchTemplateSpecification](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-spotfleet-fleetlaunchtemplatespecification.html). Since the properties of that are identical to the ones from [Autoscaling](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-autoscaling-autoscalinggroup-launchtemplatespecification.html) and [EC2](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-instance-launchtemplatespecification.html), my question was asking whether I should create this new class, or just have LaunchTemplateConfiguration use the existing one for EC2 instances.

Looking at the specification file, it looks like they actually call this a FleetLaunchTemplateSpecification, so perhaps I should call it that.

Does that make more sense?

Thanks!

Reply:
Thanks for pointing me to that spec document -- that's pretty useful. Some of the errors I found in the CF documentation are not present in the spec, which makes me wonder how the docs are generated...

Reply:
Yeah, the docs are not always correct. But I've been trying to rely on them for naming guidance. For the above naming try this:
```
make spec
python scripts/gen.py spec/Ec2SpotFleetSpecification.json
```
which is how I got the line:
```
        'LaunchSpecifications': ([SpotFleetLaunchSpecification], False),
```

Reply:
Ah, thanks. The LaunchSpecifications line is already in troposphere prior to this PR -- I'll use its suggestion for the new functionality, though.